refactor ReadMe file (#206)
GSG-G7/shrinc: README.md

# Shrinc  ![Build Status](https://api.travis-ci.com/GSG-G7/shrinc.png?branch=master)      

Medical App that Facilitate reaching to therapists and getting the appropriate therapy :sparkles: :sparkles:

## **Team Lead** :sunglasses:
* Abdallah Ammar 

## **Team Members** :smiley:
* Hashem Taha 
* Fadi Omar 
* Mohammad Alhallaq
* Asmaa Thabet

## **Main Problem**    :no_entry_sign:

People who are busy and can't find a therapist quickly and easily or who don't have enough money to ask therapists and get contact with them or who don't know the appropriate type of therapy.
    
## **Problem Solution** :bulb:

> Make app to facilitate user reaching to the appropirate therpay and connect with therapist in the right time with less costs.

Reviewer: ```suggestion
Make app to facilitate user reaching to the appropriate therapy and connect with therapist in the right time with less costs.
```

- As a user, I can see Glossary that explain scientific and strange word.
- As a user, I can visit a therapist-specific page to find out more about the service.

### **User who gives proper therapy to patients** :syringe: 
- As a therapist, I can fill out a form to create and submit a profile.

## **User Journey**  :pencil2:

### **User who want to take a therapy** :pill:

> As a user I can determine my status by answering a Questionnaire in order to find a specific therapist that I believe he/she will be appropriate for me and I can see list of Therapies according to questionnaire result, or I can choose to search (filter) for therapists by various categories like location, cost, specialty, or even by Skype/remote possibility Or I can visit a therapist profile to get more information about him/her, or learn more about therapists types, and I can see Glossary that explains scientific and strange words and I can move freely using the menu in the app, also I can get more about the app using the description that written in seprate about section.

Reviewer: ```suggestion
As a user I can determine my status by answering a Questionnaire in order to find a specific therapist that I believe he/she will be appropriate for me and I can see list of Therapies according to questionnaire result, or I can choose to search (filter) for therapists by various categories like location, cost, specialty, or even by Skype/remote possibility Or I can visit a therapist profile to get more information about him/her, or learn more about therapists types, and I can see Glossary that explains scientific and strange words and I can move freely using the menu in the app, also I can get more about the app using the description that written in separate about section.
```

### **User who gives proper therapy to patients** :syringe:
As a therapist I can sign up in the app as a a therapy by filling the form that let me enter full details about me to let the people see my profile and contact with me. 

## **Prototype** :iphone: 
[View Prototype](https://www.figma.com/proto/7ZHosfDL29DQFe7GxPcotT/Shrinc?node-id=1%3A4&scaling=contain) :eyes: :eyes:

![shrinc](https://user-images.githubusercontent.com/29041512/69190823-e7475d80-0b29-11ea-9fd6-782ff9f108b9.png)

## **Technologies**  :computer:
BackEnd: **Node JS** & **Express JS**    
FrontEnd: **React JS**  
Database: **Airtable**  
Styling:  **CSS**   
Libraries: **AntDesign** & **Leaflet**

## **Launch App**  :mega:

> First clone this repo:  
`git clone https://github.com/GSG-G7/shrinc.git`

> > Run `npm i` to install the dependencies for the app as general.

Reviewer 2: `npm i` install the dependencies and dev dependencies

> > Run `cd client` and `npm i` to install dependencies for client- React Js . 

Reviewer 2: ```suggestion
> Run `cd client` and `npm i` to install the packages for client- React Js . 
```

### Database Setup  :clipboard:

#### setup Airtable :triangular_ruler:
* You should login to  [Airtable](https://airtable.com) Account to have access to its API .    
* Click on Shrinc Workspace that it will redirect you to its Documentation API .    
* In The Documentation you will find [Airtable_KEY ], [Base_ID] environmental variables needed to run app.
* Set the database variables in your .env file as follows

#### setup Cloudinary :star:
* You should login to [Cloudinary](https://cloudinary.com/) Account .
* You will gain [Cloud_Name], [Cloudinary_KEY], [Cloudinary_SECRET] environmental variables needed to upload images to database.
* Set the database variables in your .env file as follows

### Environment Variables :key:
Environment variables are one of the ways we keep our product safe. If you want to access our app locally you will need to add your own.

First create a [.env](https://github.com/dwyl/env2#create-a-env-file) file in the App route and add the following variables:
```
Airtable_KEY
Base_ID
Cloud_Name
Cloudinary_KEY
Cloudinary_SECRET
```

### Start the App :electric_plug:

> To start the App Locally you have to start server First then start client side !

Reviewer 2: The server should not be the first !!!

> To run the Server , In your terminal Type: 

    `npm run dev`

    then you should be able to go to [localhost](http://localhost:5000/) 
> To run the client side , In your terminal Type: 
    `cd client`
    `npm start`

    then you will be able to run [localhost](http://localhost:3000/) and view the app live in the Browser!

> ## **Rescources**  :mag:

Reviewer: ```suggestion
## **Resources**  :mag:
```

* [Node Js](https://nodejs.org/en/)
* [Express Framework](https://expressjs.com/)

> * [Joi library](https://www.npmjs.com/package/joi) 

Reviewer 2: We Yup instead of Joi!!

Author: I fixed it

### **User who gives proper therapy to patients** :syringe:
As a therapist I can sign up in the app as a a therapy by filling the form that let me enter full details about me to let the people see my profile and contact with me. 

## **Prototype** :iphone: 
[View Prototype](https://www.figma.com/proto/7ZHosfDL29DQFe7GxPcotT/Shrinc?node-id=1%3A4&scaling=contain) :eyes: :eyes:

![shrinc](https://user-images.githubusercontent.com/29041512/69190823-e7475d80-0b29-11ea-9fd6-782ff9f108b9.png)

## **Technologies**  :computer:
BackEnd: **Node JS** & **Express JS**    
FrontEnd: **React JS**  
Database: **Airtable**  
Styling:  **CSS**   
Libraries: **AntDesign** & **Leaflet**

## **Launch App**  :mega:

> First clone this repo:  

> `git clone https://github.com/GSG-G7/shrinc.git`

Reviewer 2: ```suggestion
> Or write this in your terminal:
`git clone https://github.com/GSG-G7/shrinc.git`
```

> Run `npm i` to install the dependencies and dev dependencies for the app as general.

> > Run `cd client` and `npm i` to install the packages for client- React Js . 

Reviewer 3: ```suggestion
> Run `cd client` and `npm i` to install the packages for client- React Js.
```

### Database Setup  :clipboard:

#### setup Airtable :triangular_ruler:

> * You should login to  [Airtable](https://airtable.com) Account to have access to its API .    

Reviewer 3: ```suggestion
* You should login to  [Airtable](https://airtable.com) Account to have access to its API. 
```

> * Click on Shrinc Workspace that it will redirect you to its Documentation API .    

Reviewer 3: ```suggestion
* Click on Shrinc Workspace that it will redirect you to its Documentation API.    
```

* In The Documentation you will find [Airtable_KEY ], [Base_ID] environmental variables needed to run app.

> * Set the database variables in your .env file as follows

Reviewer 3: ```suggestion
* Set the database variables in your .env file as follows.
```

#### setup Cloudinary :star:

> * You should login to [Cloudinary](https://cloudinary.com/) Account .

Reviewer 3: ```suggestion
* You should login to [Cloudinary](https://cloudinary.com/) Account.
```

* You will gain [Cloud_Name], [Cloudinary_KEY], [Cloudinary_SECRET] environmental variables needed to upload images to database.

> * Set the database variables in your .env file as follows

Reviewer 3: ```suggestion
* Set the database variables in your .env file as follows.
```

### Environment Variables :key:
Environment variables are one of the ways we keep our product safe. If you want to access our app locally you will need to add your own.

First create a [.env](https://github.com/dwyl/env2#create-a-env-file) file in the App route and add the following variables:
```

> Airtable_KEY

Reviewer 4: This is a personal API key. It’s required in order to use the Airtable API.

> Base_ID

Reviewer 4: This Base_ID to define which base you want to connect with it.

> Cloud_Name

Reviewer 4: The cloud name identifier is the name of your Cloudinary account, This name is assigned to your account during signup (the field can be edited) and is listed in the Account Details.

Cloudinary_KEY

> Cloudinary_SECRET

Reviewer 4: Cloudinary auto-generates a pair of access identifiers, the API Key and API Secret, that need to be used for communicating with Cloudinary's API and signing your requests.
These identifiers are listed in the Account Details.

```

### Start the App :electric_plug:

To start the App Locally you can start server First then start client side or vice versa!

> > To run the Server , In your terminal Type: 

Reviewer 3: ```suggestion
> To run the Server, In your terminal Type: 
```

    `npm run dev`

    then you should be able to go to [localhost](http://localhost:5000/) 

> > To run the client side , In your terminal Type: 

Reviewer 3: ```suggestion
> To run the client side, In your terminal Type: 
```

    `cd client`
    `npm start`

> 

Reviewer 3: ```suggestion
```
    cd client
    npm start
```
```

BackEnd: **Node JS** & **Express JS**    
FrontEnd: **React JS**  
Database: **Airtable**  
Styling:  **CSS**   
Libraries: **AntDesign** & **Leaflet**

## **Launch App**  :mega:

> First clone this repo by typing this command in the terminal:  
`git clone https://github.com/GSG-G7/shrinc.git`

> Run `npm i` to install the packages for the app as general.

> Run `cd client` and `npm i` to install the packages for client- React Js. 

### Database Setup  :clipboard:

#### setup Airtable :triangular_ruler:
* You should login to  [Airtable](https://airtable.com) Account to have access to its API.    
* Click on Shrinc Workspace that it will redirect you to its Documentation API.    
* In The Documentation you will find [Airtable_KEY ], [Base_ID] environmental variables needed to run app.
* Set the database variables in your .env file as follows.

#### setup Cloudinary :star:
* You should login to [Cloudinary](https://cloudinary.com/) Account.
* You will gain [Cloud_Name], [Cloudinary_KEY], [Cloudinary_SECRET] environmental variables needed to upload images to database.
* Set the database variables in your .env file as follows.

### Environment Variables :key:

> Environment variables are one of the ways we keep our product safe. If you want to access our app locally you will need to add your own.

Reviewer 3: ```suggestion
Environment variables are one of the ways we keep our product safe, If you want to access our app locally you will need to add your own.
```
